src/openforms/registrations/tasks.py

```diff
@@ -118,18 +118,23 @@ def register_submission(submission_id: int) -> Optional[dict]:
         return
     # unexpected exceptions should fail the entire chain and show up in error monitoring
     except Exception as e:
-        logger.debug(
+        logger.error(
             "Registration using plugin '%r' for submission '%s' unexpectedly errored",
             plugin,
             submission,
+            exc_info=True,
         )
         submission.save_registration_status(
             RegistrationStatuses.failed, {"traceback": traceback.format_exc()}
         )
         logevent.registration_failure(submission, e, plugin)
-        raise
+        # if we're inside the retry workflow, continued failures should abort the entire
+        # chain so downstream tasks don't run with incorrect/outdated/missing data
+        if is_retrying:
```

Comment:
but they will recover if/when the root cause is solved (missing config added, code bugfix applied). it's not really possible to programmatically determine whether the failure was because of a programming error, or a remote system not being available.